### PR TITLE
feature(sct-builders): use aws ASGs as Azure builders

### PR DESCRIPTION
**Pull request by fruch**
since we don't really need the jenkins builder todo much we can use AWS ones to spin up an Azure pipeline/job

## Testing

- [x] - artifact - https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/fruch/job/artifacts-azure-image-test/4/
- [x] - longevity - https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/fruch/job/longevity-10gb-3h-azure-test/2

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by fruch** [comment]
@soyacz I would like you thought about the two commit here, one is using AWS as builders, and one I've manage to get the Jenkins plugin to partly work (only with the default Ubuntu image, and without controlling the number of executors, i.e. one machine per job)

**Review by soyacz — APPROVED** [review]
if proven works: LGTM

File: test-cases/artifacts/azure-image.yaml [hunk 1]
```diff
@@ -17,3 +17,6 @@ nemesis_class_name: 'NoOpMonkey'
 test_duration: 60
 user_prefix: 'artifacts-azure-image'
 system_auth_rf: 1
+
+# since running from runner in AWS, we need the address to be publicly available
+intra_node_comm_public: true
```
[the remark this answers is not in the document]
Comment (fruch) [reply test-cases/artifacts/azure-image.yaml:22]: It's not enough for CQL connectivity

**Comment by fruch** [comment]
testing this again:
https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/fruch/job/artifacts-azure-image-test/19/

**Comment by fruch** [comment]
@fgelcer @Mark-Gurevich @soyacz 
I've decided using this PR, and using AWS builder for running Azure jobs.
I've retested it, please review.

**Review by fgelcer — APPROVED** [review]
LGTM

**Review by soyacz — APPROVED** [review]

**Pull request merged**